src/dagmc.cpp

```diff
@@ -816,12 +850,50 @@ int32_t next_cell(int32_t surf, int32_t curr_cell, int32_t univ)
   return univp->cell_index(new_vol);
 }
 
+extern "C" int openmc_get_dagmc_cell_ids(
+  int32_t univ_id, int32_t* ids, size_t* n)
+{
+  // make sure the universe id is a DAGMC Universe
+  const auto& univ = model::universes[model::universe_map[univ_id]];
+  if (univ->geom_type() != GeometryType::DAG) {
+    fatal_error(
+      "Universe " + std::to_string(univ_id) + " is not a DAGMC Universe!");
+  }
+
+  std::vector<int32_t> dag_cell_ids;
+  for (const auto& cell_index : univ->cells_) {
+    const auto& cell = model::cells[cell_index];
+    if (cell->geom_type() == GeometryType::DAG)
+      dag_cell_ids.push_back(cell->id_);
+  }
+  std::copy(dag_cell_ids.begin(), dag_cell_ids.end(), ids);
+  *n = dag_cell_ids.size();
+}
+
+extern "C" int openmc_dagmc_universe_get_num_cells(int32_t univ_id, size_t* n)
+{
+  // make sure the universe id is a DAGMC Universe
+  const auto& univ = model::universes[model::universe_map[univ_id]];
+  if (univ->geom_type() != GeometryType::DAG) {
+    fatal_error(
+      "Universe " + std::to_string(univ_id) + " is not a DAGMC Universe!");
```

Comment:
No need to include the `!` in the error messages. If they are in other messages added here, let's remove them.

```suggestion
      "Universe " + std::to_string(univ_id) + " is not a DAGMC Universe");
```